src/Geocoder/Provider/Chain.php

```diff
@@ -38,6 +40,9 @@ public function geocode($address)
     {
         $exceptions = [];
         foreach ($this->providers as $provider) {
+            if($provider instanceof LocaleAwareProvider && $this->getLocale() !== null){
```

Comment:
@willdurand This should not be a problem, because the chainprovider delegate each call. 
Change the locale dynamicaly isn't supported yet. And in my opinion, such functionality made this code more complex.

If someone need this functionality, just create more than one provider like `provider_de, provider_en` and so on.
I think the `Geocoder` class should be decide which provider should be used not the provider itself. A provider should be invisible for the end user. And if someone _really_ need this functionality, here you go we are in OOP so extend the class.